DjangoProject/bookmarksapp/views.py

```diff
@@ -17,12 +17,52 @@ class FolderListView(ListView):
     context_object_name = 'folders'
     template_name = 'bookmarksapp/folderlist.html'
 
+    def dispatch(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            return super(FolderListView, self).dispatch(request, *args, **kwargs)
+        else:
+            return HttpResponseRedirect(reverse('authenticationapp:login'))
+
     def get_queryset(self):
         qs = super().get_queryset()
         return qs.filter(created_by=self.request.user)
 
     def get_ordering(self):
         if 'sort' in self.request.GET:
-            return self.request.GET['sort']
+            sort = self.request.GET['sort']
+            if sort not in ['-modified', '-created', 'name']:
+                sort = 'name'
+                return sort
+            return sort
+        else:
+            return None
+
+
+class BookmarksListView(ListView):
+    model = Folders
+    context_object_name = 'bookmarks'
+    template_name = 'bookmarksapp/bookmarkslist.html'
+
+    def dispatch(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            return super(BookmarksListView, self).dispatch(request, *args, **kwargs)
+        else:
+            return HttpResponseRedirect(reverse('authenticationapp:login'))
+
+    def get_queryset(self):
+        if 'sort' in self.request.GET:
+            self.queryset = Folders.objects.get(created_by=self.request.user, slug=self.kwargs['slug']).folder
```

Comment:
Whats the purpose of this?

Reply:
i am using Folders model to get bookmarks related to it instead of bookmarks model.